File: 250225/1이 되는 순간까지/until-the-moment-I-reach-one.cpp
#include <iostream>

using namespace std;

int until_one(int n) {
    if (n == 1)
        return 0;
    if (n % 2 == 0)
        return until_one(n / 2) + 1;
    else
        return until_one(n / 3) + 1;
}

int main() {
    int n;
    cin >> n;

    cout << until_one(n);
    return 0;
}

File: 250225/1이 되는 순간까지/until-the-moment-I-reach-one.js
const fs = require("fs");

const n = Number(fs.readFileSync(0, "utf8"))

function untilOne(n) {
    if (n == 1)
        return 0;
    if (n % 2 == 0)
        return untilOne(Math.floor(n / 2)) + 1;
    return untilOne(Math.floor(n / 3)) + 1;
}

console.log(untilOne(n));
